pandas/core/categorical.py

```diff
@@ -2291,6 +2292,8 @@ def _factorize_from_iterable(values):
         cat = Categorical(values, ordered=True)
         categories = cat.categories
         codes = cat.codes
+        if len(codes) and not len(categories):
+            categories = Float64Index([])
```

Comment:
instead I would just defer to ``Index(values)``, e.g.

```
In [16]: pd.Index([np.nan, np.nan])
Out[16]: Float64Index([nan, nan], dtype='float64')

In [17]: pd.Index([None, None])
Out[17]: Index([None, None], dtype='object')

In [18]: pd.Index([pd.NaT, pd.NaT])
Out[18]: DatetimeIndex(['NaT', 'NaT'], dtype='datetime64[ns]', freq=None)
```

Reply:
I would argue though that this should instead happen *inside* the ``Category`` constructor and not special case here